src/components/Modal/Modal.scss

```diff
@@ -20,19 +22,25 @@ $modal-default-border-radius: 5px;
     &__content-aligner {
         min-width: 100%;
         min-height: 100%;
+        height: var(--g-modal-height, #{$modal-default-height});
 
         display: inline-flex;
         align-items: center;
         justify-content: center;
     }
 
     &__content-wrapper {
+        width: var(--g-modal-width, #{$modal-default-width});
+        height: var(--g-modal-height, #{$modal-default-height});
+
         margin: var(--g-modal-margin, #{variables.$modal-default-margin});
         border-radius: var(--g-modal-border-radius, #{$modal-default-border-radius});
         overflow-x: hidden;
     }
 
     &__content {
+        height: var(--g-modal-height, #{$modal-default-height});
```

Comment:
Let's apply these styles only for `&__content`:
```
width: var(--g-modal-width, #{$modal-default-width});
height: var(--g-modal-height, #{$modal-default-height});
```

Applying it to all elements could result in unexpected behaviour.

Reply:
Yes, valid point. This works for the case when I want to use vh, but it's not sufficient when using %.

Reply:
content and wrapper represent inner Modal elements, their only purpose it to center the modal, they are not for public tweaking. For the end user they should not be exposed. Why `--g-modal-width` and `--g-modal-height` is not enough?

Reply:
In the project, we [used 100% for the mobile modal](https://developer.chrome.com/blog/url-bar-resizing?hl=ru), and the styles were like this: 

```
.g-modal__content-aligner {
    height: 100%;
}

.g-modal__content-wrapper,  .g-modal__content {
    height: 100%;
    width: 100%;
}
```

We will switch to [dvh](https://caniuse.com/?search=dvh) in the project. It works correctly on mobile and does not require nested styles. I simplified CSS API